### PR TITLE
implemented momentValue for the wma.ts indicator

### PR DESCRIPTION
**Pull request by real-lazy-coder**
The WMA indicator did not have a moment value.  This PR implements this for the indicator.  All tests have been passed.

### CODE REVIEW

**Comment by real-lazy-coder** [comment]
I also added the moment value to the ADX as this was not implemented either.  I'm working on a project that needs this and would like to get it merged if possible.

**Comment by BusinessDuck** [comment]
Sure, let me time to review and it  would be merged

File: src/adx.ts [hunk 1]
```diff
@@ -74,4 +74,50 @@ export class ADX {
 
         return { adx: this.wema.nextValue(100 * (diAbs / (pDI + nDI))), pdi: pDI, mdi: nDI };
     }
+
+    momentValue(h: number, l: number, c: number) {
+        if (!this.prevClose) {
+            this.prevHigh = h;
```
Comment (BusinessDuck) [inline src/adx.ts]: Moment value calculation shouldn't affect indicator state, please dont override prevHigh/prevLow/prevClose values, these values updates only inside of nextValue (when candle is closed)

File: src/adx.ts [hunk 2]
```diff
@@ -74,4 +74,50 @@ export class ADX {
...
+            nDM = 0;
+        }
+
+        const atr = this.wma1.momentValue(getTrueRange(h, l, this.prevClose));
+        const avgPDI = this.wma2.momentValue(pDM);
+        const avgNDI = this.wma3.momentValue(nDM);
+
+        this.prevHigh = h;
+        this.prevLow = l;
+        this.prevClose = c;
```
Comment (BusinessDuck) [inline src/adx.ts:107]: same here

File: src/wma.ts [hunk 3]
```diff
@@ -40,6 +40,20 @@ export class WMA {
      * does not affect any next calculations
      */
     momentValue(value: number) {
-        // return this.prevValue + (value - this.prevValue) / this.period;
+        if (!this.buffer.filled) {
+            return
+        }
+
+        const buffer = [...this.buffer.toArray()]
```
Comment (BusinessDuck) [inline src/wma.ts]: Better performance way:

```typescript
const removedItem = this.buffer.push(value);
//...apply some calculations
// also buffer contains forEach and forEachRight properties
// then restore removed by
this.buffer.pushBack(removedItem);
```

**Comment by BusinessDuck** [comment]
Also please add unit tests for `momentValues` the returning results for these new methods should be same as nextValue (for next candle)

Example with RSI:

```typescript
 it('Moment Value', () => {
        const rsi = new RSI();

        ohlc.forEach((tick) => {
            const moment = rsi.momentValue(tick.c);
            const next = rsi.nextValue(tick.c);

            expect(moment).toEqual(next);
        });
    });
```

**Comment by real-lazy-coder** [comment]
I just updated the WMA.ts and wma.spec.ts.  I'm struggling with getting the momentValue of the adx...

File: src/wma.ts [hunk 4]
```diff
@@ -40,6 +40,25 @@ export class WMA {
      * does not affect any next calculations
      */
     momentValue(value: number) {
-        // return this.prevValue + (value - this.prevValue) / this.period;
+        const buffer = new CircularBuffer(this.buffer.length);
```
Comment (BusinessDuck) [inline src/wma.ts:43]: please no :( 

just mutate original buffer and then restore it, it synchronous operation and does not have any side effects

Comment (BusinessDuck) [reply src/wma.ts:43]: push + pushback isn't work?

File: src/adx.ts (hunk 2 above)
Comment (BusinessDuck) [reply src/adx.ts:107]: remove it please, moment value calculates relates to previous close candle, not to previous tick

**Comment by BusinessDuck** [comment]
Ping me when you have done, then i can merge your MR and apply some fixes, that's works much faster than review discuss :)

**Comment by BusinessDuck** [comment]
Okay, i'll do it by myself, looks like buffer need some refactoring....

**Pull request merged**